### ISSUE TITLE
[collapsible] Layout shift when using CSS transitions and the `Panel` is initially open

### ISSUE THREAD

**Issue by mj12albert** [mui/base-ui#740]
only the one using CSS transitions is consistently flagged for CLS here

https://deploy-preview-577--base-ui.netlify.app/experiments/collapsible-cls

![Screenshot 2024-10-25 at 9 30 35 PM](https://github.com/user-attachments/assets/9f22bbb9-a814-4f85-b147-0ec4267bf685)

**Comment by mj12albert** [comment]
testing the `/components/react-menu` page:

master branch vs https://github.com/mui/base-ui/pull/577, there is basically no difference

The Demo collapsible uses CSS animations and doesn't get flagged for CLS

master
![Screenshot 2024-10-25 at 9 32 11 PM](https://github.com/user-attachments/assets/6111651a-8f9a-42f5-9c1b-086b5f27cfc6)

after changes
![Screenshot 2024-10-25 at 9 33 20 PM](https://github.com/user-attachments/assets/7b8930db-9ad2-4a1b-9bec-81cc63c1b181)

**Comment by mj12albert** [comment]
@michaldudak Do you still reproduce this on the Demo component's Collapsible? I don't get it even on `master` anymore, but using CSS transitions, the minimal example in [experiments](https://deploy-preview-577--base-ui.netlify.app/experiments/collapsible-cls) is quite consistent

**Comment by michaldudak** [comment]
No, it seems it somehow was fixed in the meantime. Lighthouse doesn't report layout shift problems anymore on master. There still is a layout shift when using transitions, but it's not flagged as an issue.
